client/blocks/product-selector/index.jsx

```diff
@@ -192,9 +192,12 @@ export class ProductSelector extends Component {
 
 		const expiryMoment = purchase.expiryDate ? moment( purchase.expiryDate ) : null;
 
+		const renewDateMoment = purchase.renewDate ? moment( purchase.renewDate ) : null;
```

Comment:
Should this actually be something more like:
```
const renewDateMoment = ! isExpiring( purchase ) && purchase.renewDate ? moment( purchase.renewDate ) : null;
```

So that it actually checks the auto-renew status?

The `isExpiring()` check is what is used elsewhere (e.g. on the `/me/purchases` page) so this would ensure consistency with that.

While testing, I actually did manage to stumble upon a scenario where the `/me/purchases` page said auto-renew was off but the code here (incorrectly) acted like it was on.  I could not figure out how to reproduce it again (nor was it obvious to me how it happened, after looking at the server-side code), but somehow it did, so I guess it's worth protecting against.

Reply:
Thanks @DavidRothstein. I didn't encounter this edge case while testing and assumed that `renewDate` sent from the backend will be empty if the auto-renew is off.

But it's better to be safe, so I've added this check.